### PR TITLE
Add tag property to horizontal navigation

### PR DESCRIPTION
**Pull request by georgegillams**
+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-ios/blob/master/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] `UNRELEASED.md`
+ [x] `README.md`
+ [x] Tests
+ [x] Adding a component? Remember to expose it in the [main `Backpack.h` header file](https://github.com/Skyscanner/backpack-ios/tree/master/Backpack/Backpack.h)
+ [x] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/master/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: Backpack/HorizontalNavigation/Classes/BPKHorizontalNavigation.m [hunk 1]
```diff
@@ -132,9 +132,10 @@ - (void)adjustBarConstraintsForView:(UIView *)view {
 - (void)updateSelection:(UIButton *)sender {
     NSInteger newIndex = [self.stackView.arrangedSubviews indexOfObject:sender];
     [self setSelectedItemIndex:newIndex];
+    BPKHorizontalNavigationItem *navItem = (BPKHorizontalNavigationItem *)sender;
```
Comment (georgegillams) [inline Backpack/HorizontalNavigation/Classes/BPKHorizontalNavigation.m]: Should be done more safely!

File: Backpack/HorizontalNavigation/Classes/BPKHorizontalNavigationDelegate.h [hunk 2]
```diff
@@ -31,7 +31,9 @@ NS_ASSUME_NONNULL_BEGIN
  * @param horizontalNavigation The `BPKHorizontalNavigation` which has been interacted with.
  * @param item The option selected by the user.
  */
-- (void)horizontalNavigation:(BPKHorizontalNavigation *)horizontalNavigation didSelectItem:(NSInteger)index;
```
Comment (k0nserv) [inline Backpack/HorizontalNavigation/Classes/BPKHorizontalNavigationDelegate.h]: The docs are out of date

File: Backpack/HorizontalNavigation/Classes/BPKHorizontalNavigationOption.h [hunk 3]
```diff
@@ -25,9 +25,12 @@ NS_ASSUME_NONNULL_BEGIN
 @interface BPKHorizontalNavigationOption : NSObject
 
 @property(nonatomic, readonly, strong, nullable) BPKIconName iconName;
+@property(nonatomic, readonly) NSInteger tag;
```
Comment (k0nserv) [inline Backpack/HorizontalNavigation/Classes/BPKHorizontalNavigationOption.h]: What's the default value for `tag` if you don't use one of the constructors that sets it. Can we use `0` to align with `UIView.tag`?

Comment (georgegillams) [reply Backpack/HorizontalNavigation/Classes/BPKHorizontalNavigationOption.h]: The default is `0` for a UIView that hasn't had tag set 👍

Comment (k0nserv) [reply Backpack/HorizontalNavigation/Classes/BPKHorizontalNavigationOption.h]: But we set it unconditionally when creating the horizontal item. We never set the value of `tag` in `BPKHorizontalNavigationOption` when a constructor without tag is used

Comment (georgegillams) [reply Backpack/HorizontalNavigation/Classes/BPKHorizontalNavigationOption.h]: Yeah but the tag value that is handed back comes from the UIView.

Comment (georgegillams) [reply Backpack/HorizontalNavigation/Classes/BPKHorizontalNavigationOption.h]: And the default value for an NSInteger is `0` anyway

Comment (k0nserv) [reply Backpack/HorizontalNavigation/Classes/BPKHorizontalNavigationOption.h]: > And the default value for an NSInteger is 0 anyway
This is not true, `NSIntger` in fact any primitive type is not guranteed to have a stable or deterministic value

File: Backpack/HorizontalNavigation/Classes/BPKHorizontalNavigationDelegate.h (hunk 2 above)
Comment (k0nserv) [reply Backpack/HorizontalNavigation/Classes/BPKHorizontalNavigationDelegate.h]: The `item` param is outdated in the doc string still

**Pull request merged**